test/azure/lro.ts

```diff
@@ -28,294 +29,186 @@ describe('typescript', function () {
     var testClient = new AutoRestLongRunningOperationTestService(credentials, baseUri, clientOptions);
     var product: AutoRestLongRunningOperationTestServiceModels.Product = { location: 'West US' };
 
-    it('should work with Put201CreatingSucceeded200', function (done) {
-      testClient.lROs.put201CreatingSucceeded200({ product: product }, function (error, result) {
-        should.not.exist(error);
-        done();
-      });
+    it('should work with Put201CreatingSucceeded200', async () => {
```

Comment:
IMO don't go out of your way to change `function` syntax to `() => ...` syntax--especially because mocha tests depend on the dynamic `this` to make calls like `.timeout` and `.skip` on a per-test or conditional basis.

Reply:
However, thank you for updating these to use Promises! This tends to make the test fail in a timely manner unlike the callbacks where we throw AssertionErrors into the aether instead of calling `done(err)`